halo2_proofs/src/plonk.rs

```diff
@@ -47,14 +48,82 @@ pub struct VerifyingKey<C: CurveAffine> {
     cs_degree: usize,
     /// The representative of this `VerifyingKey` in transcripts.
     transcript_repr: C::Scalar,
+    selectors: Vec<Vec<bool>>,
```

Comment:
I think this should be the compressed selectors. There's no reason to keep the uncompressed selectors in the verifying key, and doing so makes the representation dependent on the selector compression algorithm when it needn't be.

Reply:
I see that I said the same thing in https://github.com/zcash/halo2/issues/449#issuecomment-1031035632 . I'm nothing if not consistent, even though I'd completely forgotten making that comment!

Reply:
I agree with you. In retrospect serialization should definitely use the compressed selectors. Ideally `VerifyingKey::read` should not need to know the type of `ConcreteCircuit` or even be supplied with `params` (the latter is only used to generate `domain, cs` and get `k, n` I believe). 

Right now I have to create a `ConcreteCircuit` ahead of time and then trick Rust to do some type inference to call `read`.